packages/inngest/src/components/connect/index.ts

```diff
@@ -550,45 +563,76 @@ class WebSocketWorkerConnection implements WorkerConnection {
     });
 
     const connectTimeout = setTimeout(() => {
-      this._excludeGateways.add(startResp.gatewayGroup);
+      this.excludeGateways.add(startResp.gatewayGroup);
       rejectWebsocketConnected?.(
         new ReconnectError("Connection timed out", attempt)
       );
     }, 10_000);
 
-    let errored = false;
-    const onConnectionError = (error: unknown) => {
-      // Don't attempt to reconnect if we're already closing or closed
-      if (
-        this.state === ConnectionState.CLOSING ||
-        this.state === ConnectionState.CLOSED
-      ) {
-        return;
-      }
+    let finalEndpoint = startResp.gatewayEndpoint;
+    if (this.options.rewriteGatewayEndpoint) {
+      const rewritten = this.options.rewriteGatewayEndpoint(
+        startResp.gatewayEndpoint
+      );
+      this.debug("Rewriting gateway endpoint", {
+        original: startResp.gatewayEndpoint,
+        rewritten,
+      });
+      finalEndpoint = rewritten;
+    }
 
-      // Only process the first error per connection
-      if (errored) {
-        return;
-      }
-      errored = true;
+    this.debug("Connecting to gateway", {
+      endpoint: finalEndpoint,
+      gatewayGroup: startResp.gatewayGroup,
+    });
 
-      this.state = ConnectionState.RECONNECTING;
-      this._excludeGateways.add(startResp.gatewayGroup);
+    const ws = new WebSocket(finalEndpoint, ["v0.connect.inngest.com"]);
```

Comment:
I know it's not really from this PR, but can `"v0.connect.inngest.com"` be a const somewhere?